dace/transformation/subgraph/stencil_tiling.py

```diff
@@ -0,0 +1,640 @@
+
+            # now do mapping data name -> outer range
+            # and from that infer mapping variable -> outer range
+            local_ranges = {dn: None for dn in coverage[map_entry][1].keys()}
+            for data_name, cov in coverage[map_entry][1].items():
+                local_ranges[data_name] = subsets.union(local_ranges[data_name],
+                                                        cov)
+                # now look at proceeding maps
+                # and union those subsets -> could be larger with stencil indent
+                for child_map in children_dict[map_entry]:
+                    if data_name in coverage[child_map][0]:
+                        local_ranges[data_name] = subsets.union(
+                            local_ranges[data_name],
+                            coverage[child_map][0][data_name])
+
+            # final assignent: combine local_ranges and variable_mapping
+            # together into inferred_ranges
+            inferred_ranges[map_entry] = {p: None for p in map.params}
+            for data_name, ranges in local_ranges.items():
+                for param, r in zip(variable_mapping[data_name], ranges):
+                    # create new range from this subset and assign
+                    rng = subsets.Range((r, ))
+                    if param:
+                        inferred_ranges[map_entry][param] = subsets.union(
+                            inferred_ranges[map_entry][param], rng)
+
+        # get parameters -- should all be the same
+        params = next(iter(map_entries)).map.params.copy()
+        # define reference range as inferred range of one of the sink maps
+        self.reference_range = inferred_ranges[next(iter(sink_maps))]
+        if self.debug:
+            print("StencilTiling::Reference Range", self.reference_range)
+        # next up, search for the ranges that don't change
+        invariant_dims = []
+        for idx, p in enumerate(params):
+            different = False
+            if self.reference_range[p] is None:
+                invariant_dims.append(idx)
+                warnings.warn(
+                    f"StencilTiling::No Stencil pattern detected for parameter {p}"
+                )
+                continue
+            for m in map_entries:
+                if inferred_ranges[m][p] != self.reference_range[p]:
+                    different = True
+                    break
+            if not different:
+                invariant_dims.append(idx)
+                warnings.warn(
+                    f"StencilTiling::No Stencil pattern detected for parameter {p}"
+                )
+        
+        # during stripmining, we will create new outer map entries
+        # for easy access
+        self._outer_entries = set()
+        # with inferred_ranges constructed, we can begin to strip mine
+        for map_entry in map_entries:
+            # Retrieve map entry and exit nodes.
+            map = map_entry.map
+
+            stripmine_subgraph = {
```

Comment:
for all transformations (StripMining, MapCollapse, MapExpansion, MapToForLoop, DetectLoop), consider using the `<TransformationType>.apply_to` API (with save=False because it's already called inside a transformation). This can considerably shorten some of the code below. Not necessary but could be nice.

Reply:
👍